src/batch.rs

```diff
@@ -107,6 +108,18 @@ impl Batch {
         max
     }
 
+    fn detect_date_lenght(&self, date_flag: DateFlag) -> usize {
```

Comment:
Same here. All the variables inside the `Flags` structs will be soon private and replaced by getter functions used to wrap the logic around the flags like the priority between them.